### PR TITLE
Dropdown for seeing docs as latest or main 

### PR DESCRIPTION
**Pull request by aliabd**
Allows users to see the docs for both main and latest release. Live on [staging](https://www.gradio-staging.com/docs/). 

Workflow: 
* Cronjob on the instance pulls from github every minute to see if there are any changes and if there are it builds the website using the Dockerfile 
* Dockerfile creates a gradio wheel and uploads it to s3, also uploads all the demos to spaces (with the suffix `_main`) and adds the wheel to their requirements. It also uploads all the demos on spaces to use the latest gradio version if a new one exists. 
* Dockerfile builds the website and the docs. The `/docs` page points to a static file that is already generated and saved in the repo named `v{X.X.X}_template.html `. The `/docs/main` page is built on every commit. There's a dropdown on each page for readers to switch between them. 
* When a new version is released (`version.txt` is changed in main), a github action opens a PR to add the new versioned docs (new `v{X.X.X}_template.html ` file).  The action installs the new version and builds the docs, so there should not be any conflicts, it also uses the versioned gradio.js instead of the built one in `/docs/main`, so it basically behaves entirely like a snapshot of the docs when the version was released. 

I tested the github action on a fork of the repo. You can see an example of it working [here](https://github.com/aliabd/gradio/pull/11) or failing [here](https://github.com/aliabd/gradio/actions/runs/3392712879) because the version was a prerelease/not latest (we don't want to show docs for prereleases)

Closes: #1650

### CODE REVIEW

File: website/homepage/upload_demos.py
Comment (freddyaboulton) [inline website/homepage/upload_demos.py:51]: I think we need a newline here:

https://huggingface.co/spaces/gradio/fraud_detector_main/blob/main/requirements.txt

Comment (aliabd) [reply website/homepage/upload_demos.py]: good catch thanks! fixed: https://huggingface.co/spaces/gradio/fraud_detector_main/blob/main/requirements.txt

File: .github/workflows/build-version-docs.yml
Comment (freddyaboulton) [inline .github/workflows/build-version-docs.yml:35]: We install from pip here right? And the reason is so that we do not use the local front-end code?

File: website/check_version.py
Comment (freddyaboulton) [inline website/check_version.py:14]: nit: Maybe it's worth putting this in its own helper as I see this logic is in separate files?

File: website/homepage/src/docs/__init__.py
Comment (freddyaboulton) [inline website/homepage/src/docs/__init__.py:125]: We don't need to remove the old version builds right? Might be good to keep them for just-in-case even though there are not displayed on the website?

File: CHANGELOG.md
Comment (abidlabs) [inline CHANGELOG.md:25]: Let's move this up to 3.9 section

**Review by freddyaboulton — APPROVED** [review]
@aliabd Thank you for this useful PR! I reviewed the upload demo and buiild logic and it looks good to me. Left some questions but nothing blocking merge.

**Comment by abidlabs** [comment]
LGTM @aliabd! One thing that is happening for me is that the select dropdown to toggle between `v3.9` and `main` sometimes takes a while to load and in the meantime I see this empty dropdown:

image

**Comment by freddyaboulton** [comment]
Also these two spaces are not building because they specify incompatible gradio versions in the requirements.txt

Not sure if they are needed or we can delete.
image

**Comment by abidlabs** [comment]
I don't see any reason why we've pinned the Gradio versions in these demos. We should be able to unpin and they'll probably work fine

File: CHANGELOG.md
Comment (aliabd) [reply CHANGELOG.md]: done

File: .github/workflows/build-version-docs.yml
Comment (aliabd) [reply .github/workflows/build-version-docs.yml:35]: Not sure what you mean by local front-end code. 

We install from pip the latest released version because that's where we want to generate these docs from. Part of the docs comes from gradio as a module and not form the `/website` dir. For example `gradio.generate_documentation()`. 

But on what will show up in `/docs/main` we install gradio from the repo and not from pip (this happens inside the Dockerfile) 

Lmk if this isn't what you mean

File: website/check_version.py
Comment (aliabd) [reply website/check_version.py]: good idea, done

File: website/homepage/src/docs/__init__.py
Comment (aliabd) [reply website/homepage/src/docs/__init__.py]: yup you're right, might be useful! done

**Comment by aliabd** [comment]
@abidlabs I don't see this specific dropdown issue. Do you see that @freddyaboulton? 


https://user-images.githubusercontent.com/9021060/200069518-b96b8225-6898-4fe9-9a52-3fffb0702d7e.mov

That being said though, there is some really bad sluggishness on the website now. For example go https://gradio.app/docs and scroll really fast. you'll see it takes a while for things to render. I guess the page is just really heavy now and we'll have to look into that from the embedding side. 

No idea why those two demos pinned a gradio version to their requirements, fixed though and should auto update the spaces once this is merged.

**Comment by freddyaboulton** [comment]
@aliabd The dev server behaves the same for me. The dropdown shows both options without lag. Switching versions takes about ~1 second during which the page hangs a bit. We can probably tune that with the other website performance improvements?

**Comment by abidlabs** [comment]
Ok here's what it looks like for me:

Recording 2022-11-04 at 16 42 45

But I'm okay with getting this big one in and working on general website tune-ups in a separate PR.

**Review by abidlabs — APPROVED** [review]

**Review by aliabid94 — APPROVED** [review]
Just tested, LGTM!

**Pull request merged**
